docs: add `Bun.deepEquals` to deep-equal replacements page (#699)
es-tooling/module-replacements: docs/modules/deep-equal.md

isDeepStrictEqual(a, b) // true [!code ++]
```

## `Bun.deepEquals` (native, Bun)

Bun exposes [`Bun.deepEquals`](https://bun.com/docs/runtime/utils#bun-deepequals) globally on the Bun runtime. It accepts two values to compare and an optional `strict` flag (default `false`).

Example:

```ts
import equal from 'deep-equal' // [!code --]

const a = { foo: 'bar' }
const b = { foo: 'bar' }

equal(a, b) // true [!code --]
Bun.deepEquals(a, b) // true [!code ++]
```

For behavior equivalent to `deep-equal`'s strict mode, pass `true` as the third argument:

```ts
Bun.deepEquals(a, b, true)
```

> 

Reviewer: This should be at the end of the file

Author: Moved to end of file in 3d1f2a6.

equal(a, b) // true [!code --]
dequal(a, b) // true [!code ++]
```

## `Bun.deepEquals` (native, Bun)

> Bun exposes [`Bun.deepEquals`](https://bun.com/docs/runtime/utils#bun-deepequals) globally on the Bun runtime. It accepts two values to compare and an optional `strict` flag (default `false`).

Reviewer 2: ```suggestion
Bun has a built-in [`Bun.deepEquals`](https://bun.com/docs/runtime/utils#bun-deepequals) function. It accepts two values to compare, and an optional `strict` flag (default `false`).
```

Example:

```ts
import equal from 'deep-equal' // [!code --]

const a = { foo: 'bar' }
const b = { foo: 'bar' }

equal(a, b) // true [!code --]
Bun.deepEquals(a, b) // true [!code ++]

> ```
> 
> For behavior equivalent to `deep-equal`'s strict mode, pass `true` as the third argument:
> 
> ```ts
> Bun.deepEquals(a, b, true)
> ```

Reviewer 2: ````suggestion
equal(a, b) // true [!code --]
Bun.deepEquals(a, b) // true [!code ++]

// Strict Mode
equal(a, b, { strict: true } // true [!code --]
Bun.deepEquals(a, b, true) // true [!code ++]
```
````
